### ISSUE TITLE
[Ruby] Fix code folding for SCE_RB_STRING_QW

### ISSUE THREAD

**Issue by zufuliu** [ScintillaOrg/lexilla#132]
Current code fails for following ruby code (from https://docs.ruby-lang.org/en/master/syntax/literals_rdoc.html#label-25w+and+-25W-3A+String-Array+Literals)
```ruby
%W(#{1 + 1})
``` 

Which seems can be fixed with this change (remove `else`)
```diff
diff --git a/lexers/LexRuby.cxx b/lexers/LexRuby.cxx
index af585756..96e32a5f 100644
--- a/lexers/LexRuby.cxx
+++ b/lexers/LexRuby.cxx
@@ -1895,7 +1895,8 @@ static void FoldRbDoc(Sci_PositionU startPos, Sci_Position length, int initStyle
         } else if (style == SCE_RB_STRING_QW) {
             if (stylePrev != style) {
                 levelCurrent++;
-            } else if (styleNext != style) {
+            }
+            if (styleNext != style) {
                 levelCurrent--;
             }
         }
```

**Comment by nyamatongwe** [comment]
To clarify, it is the end character `)` that is causing the problem since it transitions both to and from `SCE_RB_STRING_QW`.

The intention of the `SCE_RB_STRING_QW` case is to fold multiline array literals like this example.
~~~ruby
keywords = %w[
__FILE__ and def end in or self unless __LINE__ begin
defined? ensure module redo super until BEGIN break do false next rescue
then when END case else for nil retry true while alias class elsif if
not return undef yield
]
puts(keywords.join('|'))
~~~

**Comment by nyamatongwe** [comment]
Included in 5.2.3.
